rosetta-source/src/main/rosetta/event-common-type.rosetta

```diff
@@ -433,88 +399,90 @@ type Trade: <"Defines the output of a financial transaction between parties - a
-                and tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> physicalSettlementTerms -> deliverableObligations -> accruedInterest exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> generalTerms -> referenceInformation exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> physicalSettlementTerms exists
+        then product -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> settlementCurrency exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> physicalSettlementTerms -> physicalSettlementPeriod exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> physicalSettlementTerms -> escrow exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> settlementTerms -> physicalSettlementTerms -> deliverableObligations -> accruedInterest exists
 
     condition FpML_cd_32: <"FpML validation rule cd-32 - If condition LongForm is true, and if fixedAmountCalculation/calculationAmount exists, then fixedAmountCalculation/dayCountFraction must exist.">
         if contractDetails -> documentation -> legalAgreementIdentification -> agreementName -> masterConfirmationType is absent
                 and contractDetails -> documentation -> legalAgreementIdentification -> agreementName -> contractualMatrix is absent
-                and tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> creditDefaultPayout -> generalTerms -> referenceInformation exists
-                and tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout -> priceQuantity exists
-                and tradableProduct -> tradeLot -> priceQuantity -> quantity -> value exists
-        then tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout -> dayCountFraction exists
+                and product -> economicTerms -> payout -> creditDefaultPayout -> generalTerms -> referenceInformation exists
+                and product -> economicTerms -> payout -> interestRatePayout -> priceQuantity exists
+                and tradeLot -> priceQuantity -> quantity -> value exists
+        then product -> economicTerms -> payout -> interestRatePayout -> dayCountFraction exists
 
     condition FpML_ird_8: <"FpML validation rule ird-8 - If the same party is specified as the payer and receiver, then different accounts must be specified.">
-        if tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout exists
-        then FpmlIrd8(tradableProduct, account) = True
+        if product -> economicTerms -> payout -> interestRatePayout exists
+        then FpmlIrd8(CreateTradableProduct(item),account) = True
```

Comment:
Updated `FpmlIrd8` type definition; @hugohills-regnosys this will need refactoring of the backend implementation.

I note separately that this function uses the attribute `account` on `Trade` which is deprecated.